AppCenter/AppCenter/Internals/Channel/MSACChannelGroupDefault.m

```diff
@@ -229,8 +229,8 @@ - (void)pauseWithIdentifyingObject:(id<NSObject>)identifyingObject {
   // Disable ingestion, sending log will not be possible but they'll still be stored.
   [self.ingestion setEnabled:NO andDeleteDataOnDisabled:NO];
 
-  // Pause each channel asynchronously.
-  for (id<MSACChannelProtocol> channel in self.channels) {
+  NSArray *copyArray = [self.channels copy];
```

Comment:
Didn't `@synchronized (self.channels)` work properly?

Reply:
No, you can check the test i wrote, it fails if we use synchronized